pkg/provider/azure_loadbalancer.go

```diff
@@ -3810,7 +3810,7 @@ func getMostEligibleLBForService(
 	ruleCount := 301
 	if existingLBs != nil {
 		for _, existingLB := range *existingLBs {
-			if StringInSlice(pointer.StringDeref(existingLB.Name, ""), eligibleLBs) {
+			if StringInSlice(trimSuffixIgnoreCase(pointer.StringDeref(existingLB.Name, ""), consts.InternalLoadBalancerNameSuffix), eligibleLBs) {
```

Comment:
We need to ignore the -internal suffix when comparing lb names, or we will ignore all of the internal lbs here.